indico/modules/events/fields.py

```diff
@@ -123,13 +123,23 @@ def _get_person_link(self, data):
                                                    'address', 'phone', '_title', 'display_order'))
         email = data.get('email', '').lower()
         if email != person_link.email:
-            if not self.event or not self.event.persons.filter_by(email=email).first():
+            existing_person = self.event.persons.filter_by(email=email).first()
+            if not self.event or not existing_person:
                 person_link.person.email = email
                 person_link.person.user = get_user_by_email(email)
                 if inspect(person).persistent:
                     signals.event.person_updated.send(person_link.person)
             else:
-                raise UserValueError(_('There is already a person with the email {email}').format(email=email))
+                # Update the person link of the existing event person with
+                # the new data and delete the old person link
+                existing_person_link = (
+                    self.person_link_cls.query.filter_by(person=existing_person, object=self.object).first() or
+                    self.person_link_cls(person=existing_person))
+                existing_person_link.populate_from_dict(data, keys=('first_name', 'last_name', 'affiliation',
+                                                                    'affiliation_link', 'address', 'phone',
+                                                                    '_title', 'display_order'))
+                db.session.delete(person_link)
```

Comment:
I think making changes here is dangerous as form validation may fail but this would still get committed. If we have to keep this, make sure to set `self.commit = False` right after the successful validate on submit in the places where these forms are used (you can find a few places where we already do that)

Reply:
Another option which is maybe less nice but definitely safer is to return the event person id in the email checker response.
Then we'd just overwrite the old person id in the person link form.